kyo-core/shared/src/main/scala/kyo/typeMap.scala

```diff
@@ -10,7 +10,23 @@ extension [A](self: TypeMap[A])
         throw new RuntimeException(s"fatal: kyo.TypeMap of contents [$self] missing value of type: [${t.show}].")
 
     inline def get[B >: A](using inline t: Tag[B]): B =
-        self.getOrElse(t, fatal).asInstanceOf[B]
+        val b = self.getOrElse(t, null)
+        if !isNull(b) then b.asInstanceOf[B]
+        else
+            var sub: B = null.asInstanceOf[B]
+            val it     = self.iterator
```

Comment:
Unfortunately this will allocate tuples, but I am not sure that's cheaply avoidable. I think a keyset iterator will still be slower.

Reply:
We can followup with some benchmarks vs ZEnvironment later.

Reply:
Makes sense. We could later explore a dedicated map implementation for this use case.